es6/experiment.js

```diff
@@ -1,213 +1,214 @@
-import Assignment from './assignment';
 import { shallowCopy, extend, isObject, forEach, map, trimTrailingWhitespace } from './lib/utils';
 
-class Experiment {
-  constructor(inputs) {
-    this.inputs = inputs;
-    this._exposureLogged = false;
-    this._salt = null;
-    this._inExperiment = true;
+export default function provideExperiment(Assignment) {
+  class Experiment {
+    constructor(inputs) {
+      this.inputs = inputs;
+      this._exposureLogged = false;
+      this._salt = null;
+      this._inExperiment = true;
 
-    this.name = this.getDefaultExperimentName();
-    this._autoExposureLog = true;
+      this.name = this.getDefaultExperimentName();
+      this._autoExposureLog = true;
 
-    this.setup();
+      this.setup();
 
-    this._assignment = new Assignment(this.getSalt());
-    this._assigned = false; 
-  }
+      this._assignment = new Assignment(this.getSalt());
+      this._assigned = false;
+    }
 
-  //helper function to return the class name of the current experiment class
-  getDefaultExperimentName() {
-    if (isObject(this) && this.constructor && this !== this.window) {
-      var arr = this.constructor.toString().match(/function\s*(\w+)/);
-      if (arr && arr.length === 2) {
-        return arr[1];
+    //helper function to return the class name of the current experiment class
+    getDefaultExperimentName() {
```

Comment:
Good catch. I think we were luckily side-stepping this issue in our usage because we provide salts and don't rely on the name discovery. I've commented on that issue with my perspective.